feat: select with cursor

Matuyuhi opened the pull request:
スワップの挙動を変更
right stick or mouseでカーソルを移動し、選択した時にアイテムがあればスワップする

Matuyuhi committed:
:sparkles: feat: Implement ISwapController interface

Matuyuhi committed:
:sparkles: feat: Added new features for cursor swapping and movement. Updated related assets and scripts.

Matuyuhi committed:
:lipstick: style: Improve readability and maintainability

Reorganized and simplified code statements, and fixed formatting and indentation. Additionally, some variable declarations were moved for better clarity.

Matuyuhi commented:
元の挙動と異なるため、仕様が決まるまでマージしない

Matuyuhi committed:
:recycle: qodana-refactor: Simplify condition check and remove unused code

ichiharayuuma committed:
キーコンフィグの変更

ichiharayuuma commented:
現状だと緑色の範囲内に複数個、アイテムが存在するとすべて選ばれた状態になりますが、
範囲内に複数のオブジェクトがあるときはよりカーソルに近いアイテムが選ばれるようにしてほしいです。

また、現状はマウスカーソルの位置で選択が行われていますが、
マウスは移動方向の制御のみ行うようにしてほしいです。
具体的には近くにアイテムがないときは一定速度でマウスの入力方向に選択範囲が移動し、
近くにアイテムがあるときはアイテムに引っ張られる方向に力を加えながら移動する感じでお願いします。
アイテムから受ける力は近いほど強く、遠いほど弱くなります。

これができたらコントローラーの対応もお願いしたいです。
キーコンフィグはRスティックでカーソルを動かし、R1トリガーでスワップ実行でお願いします。

忙しいとは思いますが実装お願いします。

Matuyuhi commented:
@ichiharayuuma 
マウスの移動方向の制御のみはどう言う意味ですか？
マウスはスティックのように入力方向を保つ動作はないと思います。頭の中で想像できないです。
あと、アイテムの位置に引っ張られるのは、ユーザーの入力方向に関係なく吸い寄せられると言うことですか？強弱あれど常に引っ張られるのであれば、常に複数アイテムからの吸い寄せでおかしな挙動になると思います。特にユーザーがカーソルを動かしていない時。

ichiharayuuma commented:
指摘ありがとうございます。
エイムアシストをつけた場合、選択範囲はカーソルとずれてしまうのでこのような表現になりました。
基本的にマウスの変位量に合わせて選択範囲を動かす形です。

エイムアシストのアルゴリズムについては今はプロトタイプなので動きができていれば中身は問いません。
僕がイメージしていたものだと、アイテムの近くで選択範囲を動かしたときにアイテム方向のベクトルを移動量に足す感じでした。

少し面倒なものだとは思いますが実装お願いします🙇

Matuyuhi commented:
あざます。それで実装します

Matuyuhi committed:
:recycle: refactor: Highlight items only in the nearest obj

Matuyuhi committed:
:sparkles: feat: Add mouse speed adjustment delta

Matuyuhi committed:
:sparkles: feat: Enhanced cursor assist navigation and game input control

Matuyuhi commented:
@ichiharayuuma 
マウスのposからdeltaの変更とアシスト実装しました。
コントローラ対応はすでにされているはずなので、何かあれば

Matuyuhi committed:
:recycle: refactor: Reorder variables and improve readability

Matuyuhi committed:
:bento: chore: Update versions of actions used in GitHub workflow

ichiharayuuma committed:
パラメーター調整、重力の適用

ichiharayuuma committed:
Merge branch 'feat/select-with-cursor' of github.com:Twilight-Studioo/prot_v1 into feat/select-with-cursor

ichiharayuuma commented:
確認しました。
実装ありがとうございます！
かなりいい感じです。

パラメーターを調整したものをプッシュしました。

コントローラーの方は手持ちのコントローラーの接触不良で試せてないので今度試します。

![LGTM](https://image.lgtmoon.dev/175890)

Matuyuhi commented:
@ichiharayuuma 
機能が完了であればマージしてリファクタ作業しますがどうですか？

ichiharayuuma commented:
マージは企画が通ってからでいいと思います。
なので今はこのままにします。

Matuyuhi committed:
:sparkles: feat: mainCamera lerp to playerPosition

Matuyuhi committed:
:recycle: refactor: Restrict editor-specific methods to UNITY_EDITOR only, remove unused code

In this commit, several methods only meant to be used in the Unity editor are now guarded with UNITY_EDITOR preprocessor directives. Additionally, unused code related to Gizmos drawing in RaycastEx and PlayerView has been removed.

Matuyuhi committed:
:bento: chore: Add GitHub workflow for game testing and building

Matuyuhi committed:
Revert ":bento: chore: Add GitHub workflow for game testing and building"

This reverts commit 30de8afbe5ba99b0c7fe819353fdf757c7f428ca.

Matuyuhi committed:
Merge branch 'main' into feat/select-with-cursor

Matuyuhi committed:
Merge branch 'main' into feat/select-with-cursor
